src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -137,163 +135,218 @@ public void clearCache() {
+      //       "read_timestamp": 123456789,
+      //       "priority": 1,
+      //       "context": "base64 encoded context"
+      //     },
+      //     {
+      //       "id": 3,
+      //       "read_timestamp": 123456789,
+      //       "priority": 2,
+      //       "context": "base64 encoded context"
+      //     }
+      //   ]
+
+      JsonNode entriesNode = rootNode.path("entries");
+      if (entriesNode != null && entriesNode.isArray()) {
+          for (JsonNode entryNode : entriesNode) {
+            QueryContextElement entry = deserializeQueryContextElement(entryNode);
+            merge(entry.id, entry.readTimestamp, entry.priority, entry.context);
           }
-        }
-      } catch (Exception e) {
-        logger.debug("deserializeFromArrowBase64: Exception = {}", e.getMessage());
-        // Not rethrowing. clear the cache as incomplete merge can lead to unexpected behavior.
-        clearCache();
       }
+    }catch (Exception e) {
+      logger.debug("deserializeQueryContextJson: Exception = {}", e.getMessage());
+      // Not rethrowing. clear the cache as incomplete merge can lead to unexpected behavior.
+      clearCache();
+    }
 
-      // After merging all entries, truncate to capacity
-      checkCacheCapacity();
+    // After merging all entries, truncate to capacity
+    checkCacheCapacity();
 
-      logger.debug("deserializeFromArrowBase64() returns");
-      // Log existing cache entries
-      logCacheEntries();
-    } // Synchronized
-  }
+    // Log existing cache entries
+    logCacheEntries();
 
-  /**
-   * Deserialize a query context entry from an Arrow vector
-   *
-   * @param root the Arrow vector root
-   * @param pos the position of the entry to be deserialized from
-   * @return a deserialized query context entry
-   */
-  private static QueryContextElement deserializeEntry(VectorSchemaRoot root, int pos) {
-    BigIntVector idVector = (BigIntVector) root.getVector(ID_POS);
-    BigIntVector tsVector = (BigIntVector) root.getVector(TIMESTAMP_POS);
-    BigIntVector priorityVector = (BigIntVector) root.getVector(PRIORITY_POS);
-    VarBinaryVector contextVector = (VarBinaryVector) root.getVector(CONTEXT_POS);
-
-    return createElement(
-        idVector.get(pos),
-        tsVector.get(pos),
-        priorityVector.get(pos),
-        contextVector.isNull(pos) ? null : contextVector.get(pos));
-  }
+  }// Synchronized
+}
 
-  /**
-   * @param id the id of the element
-   * @param timestamp the last update timestamp
-   * @param priority the priority of the element
-   * @param opaqueContext the binary data of the opaque context
-   * @return a query context element
-   */
-  private static QueryContextElement createElement(
-      long id, long timestamp, long priority, byte[] opaqueContext) {
-    return new QueryContextElement(id, timestamp, priority, opaqueContext);
-  }
+private static QueryContextElement deserializeQueryContextElement(JsonNode node) throws IOException {
```

Comment:
How is this different from `deserializeQueryContextJson`? I am confused as to why we need both when we've already deserialized JSON information received in the response object into the cache.

Reply:
This `deserializeQueryContextElement` is just a helper function to deserialize each entry